surveySimPP/modules/benchmarks/bench_ApplyColourOffsets.py

```diff
@@ -0,0 +1,104 @@
+import timeit
+import numpy as np
+import pandas as pd
+
+from memory_profiler import memory_usage
+from metric_emitter import Emitter
+
+from surveySimPP.modules.PPApplyColourOffsets import PPApplyColourOffsets
+
+
+def setup_test_obs(num_copies=1):
+    objects = np.array(['obj1', 'obj1', 'obj1', 'obj1', 'obj2', 'obj2', 'obj2', 'obj2']*num_copies)
+
+    optfilter = np.array(['r', 'g', 'i', 'z', 'r', 'g', 'z', 'i']*num_copies)
+
+    ur = np.array([0.1, 0.1, 0.1, 0.1, 0.2, 0.2, 0.2, 0.2]*num_copies)
+    gr = np.array([0.3, 0.3, 0.3, 0.3, 0.4, 0.4, 0.4, 0.4]*num_copies)
+    ir = np.array([0.5, 0.5, 0.5, 0.5, 0.6, 0.6, 0.6, 0.6]*num_copies)
+    zr = np.array([0.7, 0.7, 0.7, 0.7, 0.8, 0.8, 0.8, 0.8]*num_copies)
+
+    H = np.array([10.0, 10.0, 10.0, 10.0, 12.0, 12.0, 12.0, 12.0]*num_copies)
+    G = np.array([0.15, 0.15, 0.15, 0.15, 0.12, 0.12, 0.12, 0.12]*num_copies)
+
+    return pd.DataFrame({'ObjID': objects, 'optFilter': optfilter,
+                                    'u-r': ur, 'g-r': gr, 'i-r': ir, 'z-r': zr,
+                                    'H_r': H, 'GS': G})
+
+def setup_other_colors():
+    return ['u-r', 'g-r', 'i-r', 'z-r']
+
+def setup_obs_filters():
+    return ['r', 'u', 'g', 'i', 'z']
+
+def create_runtime_statement():
+    return 'PPApplyColourOffsets(test_obs.copy(), "HG", othercolours, observing_filters, "r")'
+
+def create_setup_str(num_copies=1):
+    return f'''
+test_obs = setup_test_obs({num_copies})
+othercolours = setup_other_colors()
+observing_filters = setup_obs_filters()
+'''
+
+class TestBenchApplyColourOffsets:
+    
+
+    '''Runtime benchmarking'''
+    def test_bench_runtime(self):
+
+        for n in [1_000, 10_000]:
+
+            emitter = Emitter(namespace="lsst.lf",
+                              name="sspp.module_benchmarks",
+                              module=PPApplyColourOffsets.__name__,
+                              benchmark_type=f'run_{n}',
+                              benchmark_unit='s')
+
+            t = timeit.Timer(
+                stmt=create_runtime_statement(),
+                setup=create_setup_str(n),
+                globals={
```

Comment:
For a self contained file like this, it might be easier to just use `globals=globals()`. But no need to change it for this PR.

Reply:
Ah, ok. That's a good call. I'll address that in a future PR.